contracts/Wallet.sol

```diff
@@ -1,16 +1,68 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.20;
 
-import {IWallet} from "./IWallet.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "./IExtendedERC20.sol";
+import "./IWallet.sol";
+import "./TokenListManager.sol";
 
 contract Wallet is IWallet {
-    // Implement getTokenMarketPrice method
-    function getTokenMarketPrice(address token) external view override returns (uint256) {
+    TokenListManager public tokenListManager;
 
+    constructor(address _tokenListManager) {
+        tokenListManager = TokenListManager(_tokenListManager);
     }
 
-    // Implement getPaginatedTokenInfo method
-    function getPaginatedTokenInfo(address user, uint start, uint limit) external view override returns (TokenInfo[] memory tokensInfo) {
+    function getPaginatedTokenInfo(address user, uint start, uint limit) 
+    external view override returns (TokenInfo[] memory tokensInfo) {
```

Comment:
Better to return `total` value for pagination view.

Reply:
In process

Reply:
Don't forget this comment, which is very important for UI side to display total info.

Reply:
Solved, I created a new structure to return tokenInfo and total.